pkg/servicemesh/controller/memberroll/controller.go

```diff
@@ -112,7 +126,21 @@ func (smmrc *serviceMeshMemberRollController) newserviceMeshMemberRollListener(l
 		currentNamespaces: nil,
 		name:              name,
 	}
-	handler.updateNamespaces("add", smmrc.memberRollName, smmrc.getNamespaces(nil))
+
+	go func() {
+		smmrLog.Debugf("Listener for %s created", name)
+		<-smmrc.cacheWarmed
```

Comment:
isn't this reading from a channel that might be long closed?

Reply:
yes, that's the idea (reading from a closed channel is a nop). The channel remains open until the cache is warmed, hence it blocks. Once channel is closed, it doesn't block anymore and the behavior is the same we have currently.

Reply:
oh yeah, you're right. so we're not leaking a goroutine, that's good. but aren't we still sending the same event twice?